cstrml/storage_order/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+//! The Substrate Node runtime. This can be compiled with `#[no_std]`, ready for Wasm.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![feature(option_result_contains)]
+
+use codec::{Decode, Encode};
+use frame_support::{decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
+    traits::{
+        Currency, ExistenceRequirement, LockableCurrency,
+	}};
+use sp_std::{str, vec::Vec};
+use system::ensure_signed;
+use sp_runtime::{traits::StaticLookup};
+use tee;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+// Crust runtime modules
+use primitives::{MerkleRoot, Balance, BlockNumber};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, Default)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct StorageOrder<T> {
+    pub file_indetifier: MerkleRoot,
+    pub file_size: u64,
+    pub expired_duration: BlockNumber,
+    pub expired_on: BlockNumber,
+    pub destination: T
+}
+
+
+/// An event handler for 
+pub trait OnOrderStroage<AccountId> {
+    fn storage_fee_transfer(transactor: &AccountId, dest: &AccountId, value: Balance) -> u64;
+}
+
+impl<AId> OnOrderStroage<AId> for () {
+    fn storage_fee_transfer(_: &AId, _: &AId, _: Balance) -> u64 {
+        // transfer the fee and return order id
+        0
+    }
+}
+
+/// The module's configuration trait.
+pub trait Trait: system::Trait + tee::Trait {
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+    type OnOrderStroage: OnOrderStroage<Self::AccountId>;
+}
+
+// TODO: add add_extra_genesis to unify chain_spec
+// This module's storage items.
+decl_storage! {
+    trait StoreOrder for Module<T: Trait> as StoreOrder {
+        pub StorageOrders get(fn storage_orders) config(): map (T::AccountId, u64) => Option<StorageOrder<T::AccountId>>; // Cannot use MerkleRoot as the key. cannot open the apps
+    }
+}
+
+// The module's dispatchable functions.
+decl_module! {
+    /// The module declaration.
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Initializing events
+        // this is needed only if you are using events in your module
+        fn deposit_event() = default;
+
+        fn store_storage_order(
```

Comment:
Maybe considering adding `TODO` on passing params using `struct` or like `this`?